icd/api/vk_descriptor_set.cpp

```diff
@@ -454,12 +467,10 @@ void DescriptorUpdate::WriteDescriptorSets(
         switch (static_cast<uint32_t>(params.descriptorType))
         {
         case VK_DESCRIPTOR_TYPE_SAMPLER:
-            if (hasImmutableSampler)
-            {
-                VK_ASSERT(!"Immutable samplers cannot be updated");
-            }
-            else
-            {
+            if(params.pImageInfo->sampler == VK_NULL_HANDLE && hasImmutableSampler) {
+              uint32_t * pSamplerDesc =(pDestSet->Layout()->Info().imm.pImmutableSamplerData + destBinding.imm.dwOffset);
+              memcpy(pDestAddr, pSamplerDesc, 4*destBinding.imm.dwSize);
```

Comment:
In the cases I was looking at, it seemed like there was a write to the immutable sampler.  I did not trace where this came from.  If it is coming from the application, then it is probably an error on their part.  If not, then there might be something wrong  in the XGL code, but I'm surprised this assert is not being hit more often.

I just want to clarify that that a descriptor write for an immutable sampler is an error.

Reply:
I think it is clear in the spec that an application updating an immutable sampler is an error:

Immutable samplers are permanently bound into the set layout and must not be changed; updating a VK_DESCRIPTOR_TYPE_SAMPLER descriptor with immutable samplers is not allowed